shaders/src/light_directional.fs

```diff
@@ -46,8 +46,10 @@ void evaluateDirectionalLight(const MaterialInputs material,
         if (hasDirectionalShadows && cascadeHasVisibleShadows) {
             uint layer = cascade;
 #if defined(HAS_VSM)
-            highp float fragDepth = length(vertex_worldPosition - frameUniforms.lightPosition.xyz);
-            visibility = shadowVsm(light_shadowMap, layer, getCascadeLightSpacePosition(cascade), fragDepth);
+            highp vec3 lightSpacePosition = getCascadeLightSpacePosition(cascade);
+            // Flip the sign of the Z coordinate; -Z is forward in light space.
+            lightSpacePosition.z *= -1.0;
```

Comment:
You should be able to include the negation in the matrix, right? (since it's already a weird matrix anyways) -- this way you wouldn't need a `#if HAS_VSM` case here.

Reply:
Yeah, I can fold that term into the matrix. I'll still need a `HAS_VSM` case, because we're calling a unique shadow function (`shadowVsm`).